[8.0][product_custom_info] New module to have custom properties in products

yajo opened the pull request:
Continuing from #120, this is only the product part, now that the base part is in https://github.com/OCA/server-tools/pull/313.

It allows to add custom attributes to products.

Obviously, this will fail until https://github.com/OCA/server-tools/pull/313 is merged.

@rafaelbn @pedrobaeza @carlos-incaser.

carlosdauden committed:
[ADD][8.0] product_custom_info: Initial development

carlosdauden committed:
[ADD][8.0] base_custom_info: View

carlosdauden committed:
[FIX][8.0] product_custom_info: Invisible field not saved

yajo committed:
Fix many things and prepare for OCA.

pedrobaeza commented on oca_dependencies.txt:
```diff
@@ -0,0 +1 @@
+server-tools
```
Put here
```
server-tools https://github.com/Yajo/server-tools custom-info
```
and you'll be able to test this PR while the other PR is without being merged.

pedrobaeza commented on product_custom_info/__init__.py:
```diff
@@ -0,0 +1,6 @@
+# © 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from . import models
```
Line at the end

pedrobaeza commented on product_custom_info/README.rst:
```diff
@@ -0,0 +1,61 @@
+
+This module extends products to allow having custom properties on them.
+
+Usage
```
Add installation section indicating the dependency on OCA/server-tools

pedrobaeza commented on product_custom_info/models/product_custom_info.py:
```diff
@@ -0,0 +1,11 @@
+
+class ProductTemplate(models.Model):
+    _name = "product.template"
+    _inherit = ["product.template", "custom.info"]
```
You should also inherit properties creating a new table, or this won't work when having multiple models that uses these tables.

Yajo replied:
Not. See https://github.com/OCA/server-tools/pull/313#discussion_r48032780.

yajo committed:
Cleanup.

yajo committed:
Better instructions.

yajo committed:
Fix model name in security file.

yajo committed:
Use lazy calculation of xmlids.

Yajo commented:
Well, that line I added in https://github.com/Yajo/product-attribute/commit/fac5eb89e85d6c721187f5ce9dee9548f2e59091 seems ignored in runbot, so not quite useful... Is that expected?

yajo committed:
Update to latest base_custom_info code.

yajo commented:
WIP no more, https://github.com/OCA/server-tools/pull/313 is merged.

I rebased and removed fac5eb89e85d6c721187f5ce9dee9548f2e59091. Let's see if it gets :green_heart:

pedrobaeza commented on product_custom_info/README.rst:
```diff
@@ -0,0 +1,66 @@
+
+* Install `OCA/server-tools <https://github.com/OCA/server-tools>`_.
+
+Usage
```
It lacks section Configuration to indicate where you create the Custom Info Templates.

pedrobaeza commented on product_custom_info/README.rst:
```diff
@@ -0,0 +1,66 @@
+Installation
+============
+
+* Install `OCA/server-tools <https://github.com/OCA/server-tools>`_.
```
Indicate the exact module it depends.

yajo committed:
Fix docs and credits.

yajo commented:
This one should be ready.

pedrobaeza commented:
I see that I cannot create by hand templates to test this, because model fields are readonly. The problem should be in base_custom_info.

yajo commented:
Well, it's the expected behavior. It's readonly to avoid users changing model of template and breaking things. You are expected to manage templates from the product form. Please follow usage instructions and you'll see it works fine.

yajo commented:
... and in any case, if you consider that a bug, it's a discussion for `base_custom_info` as you pointed.
